src/odemis/driver/xt_client.py

```diff
@@ -2674,14 +2675,16 @@ def _acquire(self):
                 self._acq_wait_start()
                 logging.debug("Preparing acquisition")
                 while True:
-                    if self._acq_should_stop():
+                    logging.debug("Start acquiring an image")
+                    if self._acq_should_stop(timeout=0.5):
+                        logging.debug("Image acquisition should stop, exiting loop")
                         break
 
                     if hasattr(self._scanner, "blanker"):
                         if self._scanner.blanker.value is None:
                             self.parent.unblank_beam()
 
-                    # Start a complete scan.
+                    logging.debug("Start a complete scan of an image")
```

Comment:
Just before starting the acquisition you should note the time, and store this information in MD_ACQ_DATE. This is used in several codes to check that the image received is indeed new enough.